compiler/rustc_monomorphize/src/partitioning.rs

```diff
@@ -161,10 +161,11 @@ where
 
     debug_dump(tcx, "INITIAL PARTITIONING", &codegen_units);
 
+    // njn: update comment
     // Merge until we have at most `max_cgu_count` codegen units.
     // `merge_codegen_units` is responsible for updating the CGU size
     // estimates.
-    {
+    if cx.tcx.sess.opts.incremental.is_none() {
```

Comment:
Currently `-Ccodegen-units` overrides the 256 default for incr comp. With this change that no longer happens. I think user's intent should be preserved even with incr comp. Maybe they care about runtime perf too and many codegen units is a huge regression (eg in a game) or maybe duplication of inlined functions costs more than not merging saves in reusable cgus. Either way the user should have the option to do what is best for them.

Reply:
Sure, as written I've done this too crudely. If this were to land various other changes would be needed, e.g. documentation. I have changed the PR to draft status to better reflect this.